### PR TITLE
Accounts fix pending

### PR DESCRIPTION
**Pull request by DenisIvanov26**
Updated and fixed account methods

### CODE REVIEW

**Review by alanorwick — CHANGES_REQUESTED** [review]

File: testingResources/keyPairTestConstants.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+export const testOldMetamaskVersion = 'MetaMask/v10.14.6-flask.0';
+export const testNewMetamaskVersion = 'MetaMask/v10.16.0-flask.1';
+
+export const testOldBip44Entropy = {
+  coin_type: 994,
+  depth: 2,
+  key: 'Owl8ESiPOy6Ti2OmB3pS3VOAOhXwU5AF4yedFmOF2iZY3SI5Po+DMB13OKZ8LdTHTWwDNzsAfzb58tJ2WkyYpA==',
+  path: "m / bip32:44' / bip32:994'",
+};
+export const testNewBip44Entropy = {
```
[the remark this answers is not in the document]
Comment (skunal8197) [reply testingResources/keyPairTestConstants.js:21]: Sorry this is an unused test file that was added as part of my merge into Denis' branch. This file will change as I make progress on unit testing

File: src/accounts.js [hunk 2]
```diff
@@ -50,14 +50,17 @@ export default class Accounts {
       };
     } else {
       console.log('have stored accounts');
+      console.log(storedAccounts);
```
Comment (skunal8197) [inline src/accounts.js:53]: Do we need these logs

File: src/accounts.js [hunk 3]
```diff
@@ -71,12 +74,14 @@ export default class Accounts {
...
     }
-    if (this.accounts.hasOwnProperty(addr)) {
-      const tempAccount = this.accounts[addr];
-      if (tempAccount.type === 'generated') {
-        const Account = await this.generateAccount(tempAccount.path);
+    for (var i = 0; i < this.accounts.length; i++) {
+      if (this.accounts[i].addr == addr) {
+        const tempAccount = this.accounts[i];
+        if (tempAccount.type === 'generated') {
+          const Account = await this.generateAccount(tempAccount.path);
```
Comment (skunal8197) [inline src/accounts.js:81]: I'd keep variable names lowercased

File: src/accounts.js [hunk 4]
```diff
@@ -97,27 +102,30 @@ export default class Accounts {
...
-        params: ['update', { currentAccountId: addr, Accounts: this.accounts }],
-      });
-      return { currentAccountId: addr, Accounts: this.accounts };
-    } else {
-      return { error: 'account not found' };
+    for (var i = 0; i < this.accounts.length; i++) {
+      if (this.accounts[i].addr == addr) {
+        this.currentAccountId = addr;
+        this.currentAccount = await this.unlockAccount(addr);
+        console.log('this.currentAccount', addr, this.currentAccount);
```
Comment (skunal8197) [inline src/accounts.js:109]: remove log

File: src/accounts.js [hunk 5]
```diff
@@ -97,27 +102,30 @@ export default class Accounts {
...
   }
 
   async getAccounts() {
+    console.log('Getting accounts');
     if (!this.loaded) {
       await this.load();
     }
-    const arrayOfObj = Object.entries(this.accounts).map((e) => e[1]);
-    return arrayOfObj;
+    console.log(this.accounts);
```
Comment (skunal8197) [inline src/accounts.js:125]: remove log

File: src/accounts.js [hunk 6]
```diff
@@ -138,24 +146,31 @@ export default class Accounts {
...
     }
-
-    console.log('accounts length', oldPath);
-
     const Account = await this.generateAccount(oldPath + 1);
     const address = Account.addr;
     const path = oldPath + 1;
-    this.accounts[address] = {
+    let context = GetShardFromAddress(address);
+    console.log('context: ', context);
```
Comment (skunal8197) [inline src/accounts.js:164]: remove log

**Review by skunal8197 — CHANGES_REQUESTED** [review]

**Comment by skunal8197** [comment]
A few call-outs
1. Testing for these changes will be done manually in accordance to [this list of functional tests](
https://www.notion.so/dominantstrategies/6-Functional-Testing-4bb4190044b74e98a4d00a136c410d9f) using the snap's frontend interface
2. The testing files that were added in this PR are a WIP and will change as we make progress on unit testing

**Comment by skunal8197** [comment]
#69 has all these changes + more so I am closing this

**Pull request closed without merging**